core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -62,4 +65,12 @@ private[history] abstract class ApplicationHistoryProvider {
    */
   def getConfig(): Map[String, String] = Map()
 
+  /**
+   * Writes out the event logs to the output stream provided. The logs will be compressed into a
```

Comment:
Sorry, messed up the doc. If the data is in the form of multiple files for each attempt (like legacy files), they will be zipped into a single one, and the single zipped file is written out.